File: 1544-make-the-string-great/1544-make-the-string-great.cs
public class Solution 
{
    public string MakeGood(string s) 
    {
        Func<char,bool> isUpper = c => c == Char.ToUpper(c);
        Stack<char> stack = new Stack<char>();
        
        foreach (char c in s)
        {
            if (stack.Count == 0)
            {
                stack.Push(c);
                continue;
            }
            if (isUpper(c))
            {
                if (stack.Peek() == Char.ToLower(c))
                {
                    stack.Pop();
                }
                else 
                {
                    stack.Push(c);
                }
            }
            else 
            {
                if (stack.Peek() == Char.ToUpper(c))
                {
                    stack.Pop();
                }
                else 
                {
                    stack.Push(c);
                }
            }
        }
        char[] res = stack.ToArray();
        Array.Reverse(res);
        return new string(res);
    }
}

File: 1544-make-the-string-great/1544-make-the-string-great.js
/**
 * @param {string} s
 * @return {string}
 */
const makeGood = function(s) {
    const isUpper = c => c === c.toUpperCase();
    const stack = [];
    
    for (const c of s) {
        if (isUpper(c)) {
            if (c.toLowerCase() === stack[stack.length - 1]) {
                stack.pop();
            }
            else stack.push(c);
        }
        else {
            if (c.toUpperCase() === stack[stack.length - 1]) {
                stack.pop();
            }
            else stack.push(c);
        }
    }
    
    return stack.join('');
};